LondonDataServices.IDecide.Core/Brokers/Securities/SecurityBroker.cs

```diff
@@ -149,5 +152,15 @@ private static ClaimsPrincipal GetClaimsPrincipalFromToken(string token)
         /// <returns>An <see cref="string"/> object containing user details.</returns>
         public async ValueTask<string> GetIpAddressAsync() =>
             await Task.FromResult(this.remoteIpAddress);
+
+        /// <summary>
+        /// Retrieves the value of a specified header from the current request.
+        /// </summary>
+        /// <param name="key">The header key.</param>
+        /// <returns>The value of the specified header, or an empty string if not found.</returns>
+        public async ValueTask<string> GetHeaderAsync(string key) =>
```

Comment:
headers is only initialised in one constructor. Ensure this will not error if one of the other constructors is used.

Add this to the non-httpcontextaccesor contructors:
`this.headers = null;`


```suggestion
    public ValueTask<string> GetHeaderAsync(string key)
    {
        if (string.IsNullOrWhiteSpace(key) || this.headers is null)
            return string.Empty;

        return ValueTask.FromResult(this.headers.TryGetValue(key, out var value)
            ? (string)value
            : string.Empty);
    }
```